### ISSUE TITLE
[BUG] OpenID Connect login nextUrl drops URL hash

### ISSUE THREAD

**Issue by wkruse** [opensearch-project/security-dashboards-plugin#1141]
**What is the bug?**
In Opensearch Dashboards URL hash encodes the state of a visualization (state could be the filters to apply, the time range etc). Currently for OpenID Connect based authentication when user logins/re-logins the url hash is being dropped because of redirecting to the IDP.

**How can one reproduce the bug?**

1. Setup OpenID Connect based authentication
2. Open any dashboard
3. Copy the URL
4. Logout
6. Open new private window
8. Paste the URL
7. You are redirected to `/app/dashboards` (dashboards overview)

**What is the expected behavior?**
The state of the dashboard/visualization must not be lost.

**What is your host/environment?**
 - Version 2.3.0

**Do you have any additional context?**
The same issue for SAML was fixed.
- #831
- #1039

**Comment by scrawfor99** [comment]
[TRIAGE] @cliu123 tagging you for follow-up with Jimish. Added to sprint backlog for reviewing the provided fix and then adapting the pull request to match new issue.

**Comment by davidlago** [comment]
@cwperks please add details to criteria including fixing Cypress in our repo to be able to fully test this change

**Comment by cwperks** [comment]
There's a few associated items that will help to make changes in the OIDC with confidence. There are unit tests in the [security repo](https://github.com/opensearch-project/security) around the OIDC Authenticator and unit tests in this repo with the OIDC flow, but there are no integration tests that test the flows with a sample OIDC Identity Provider. For SAML, there are [integration tests](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/test/jest_integration/saml_auth.test.ts) in this repo that [spin up a node-based IdP](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/package.json#L20) and use selenium to test common use-cases:

1. Login to app/opensearch_dashboards_overview#/ when SAML is enabled
2. Login to app/dev_tools#/console when SAML is enabled
3. Login to Dashboard with Hash
4. Tenancy persisted after Logout in SAML

There are no equivalent tests for OIDC, but there should be and an issue was created to track the work: https://github.com/opensearch-project/security-dashboards-plugin/issues/1150

Function tests for dashboards-plugins are typically written with Cypress in the [opensearch-dashboards-functional-test repo](https://github.com/opensearch-project/opensearch-dashboards-functional-test) and the selenium tests in this repo should migrate to the FTR (functional test repository) and be referenced in a github action like this: https://github.com/opensearch-project/security-dashboards-plugin/blob/main/.github/workflows/cypress-test.yml

There is one issue with Cypress in the FTR repo now. The FTR repo is on CypressV9 which does not support cross origin testing. There is a PR in the FTR repo to upgrade Cypress to V12 but it is still a WIP to identify tests that need to be updated: https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/668

Work has been done in the FTR repo to add OIDC tests, but it is blocked by the CypressV12 upgrade: https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/595

**Comment by derek-ho** [comment]
Closing - was fixed in: https://github.com/opensearch-project/security-dashboards-plugin/pull/1563
